backend/transform.py

```diff
@@ -25,20 +27,34 @@
 """
 
 
-def get_nodes_using_sentence_splitter(documents: list[Document], config: Dict) -> list:
+def get_nodes_using_SentenceSplitter(documents: list[Document], config: Dict) -> list:
     sentence_splitter = SentenceSplitter(**config)
-    return sentence_splitter.get_nodes_from_documents(documents)
+    nodes = sentence_splitter.get_nodes_from_documents(documents)
+    return nodes
 
 
-def get_nodes_using_sentence_window_node_parser(documents: list[Document], config: Dict) -> list:
+def get_nodes_using_SentenceWindowNodeParser(documents: list[Document], config: Dict) -> list:
     sentence_window_node_parser = SentenceWindowNodeParser(**config)
-    return sentence_window_node_parser.get_nodes_from_documents(documents)
+    nodes = sentence_window_node_parser.get_nodes_from_documents(documents)
+    return nodes
 
 
-def get_nodes_using_recursive_character_text_splitter(documents: list[Document], config: Dict) -> list:
+def get_nodes_using_RecursiveCharacterTextSplitter(documents: list[Document], config: Dict) -> list:
     recursive_character_text_splitter = LangchainNodeParser(
         RecursiveCharacterTextSplitter(**config))
-    return recursive_character_text_splitter.get_nodes_from_documents(documents)
+    nodes = recursive_character_text_splitter.get_nodes_from_documents(
+        documents)
+    return nodes
+
+
+def get_nodes_using_SemanticSplitterNodeParser(documents: list[Document], config: Dict) -> list:
+    embedding = BaseEmbedding(model_name="gemini-pro",
+                              embed_batch_size=config["embed_batch_size"])
+    semantic_splitter_node_parser = SemanticSplitterNodeParser(
+        embed_model=embedding, **config)
```

Comment:
config dictionary will still have the embed_batch_size, does passing it to the SemanticSplitterNodeParser causes any issue?